clusterloader2/pkg/measurement/common/service_creation_latency.go

```diff
@@ -0,0 +1,317 @@
+			return nil, err
+		}
+		s.waitTimeout, err = util.GetDurationOrDefault(config.Params, "waitTimeout", defaultServiceCreationLatencyTimeout)
+		if err != nil {
+			return nil, err
+		}
+		return nil, s.start()
+	case "waitForReady":
+		err = s.waitForReady()
+		if err != nil {
+			klog.Errorf("%s: %v", s, err)
+		}
+		return nil, err
+	case "gather":
+		return s.gather()
+	default:
+		return nil, fmt.Errorf("unknown action %v", action)
+	}
+}
+
+// Dispose cleans up after the measurement.
+func (s *serviceCreationLatencyMeasurement) Dispose() {
+	if s.isRunning {
+		s.isRunning = false
+		close(s.stopCh)
+	}
+	s.queue.Stop()
+	s.pingCheckers.Dispose()
+}
+
+// String returns a string representation of the metric.
+func (s *serviceCreationLatencyMeasurement) String() string {
+	return serviceCreationLatencyName + ": " + s.selector.String()
+}
+
+func (s *serviceCreationLatencyMeasurement) start() error {
+	if s.isRunning {
+		klog.Infof("%s: wait for controlled pods measurement already running", s)
+		return nil
+	}
+	klog.Infof("%s: starting wait for controlled pods measurement...", s)
+
+	s.isRunning = true
+	s.stopCh = make(chan struct{})
+
+	i := informer.NewInformer(
+		s.client,
+		"services",
+		s.selector,
+		func(oldObj, newObj interface{}) {
+			f := func() {
+				s.handleObject(oldObj, newObj)
+			}
+			s.queue.Add(&f)
+		},
+	)
+	return informer.StartAndSync(i, s.stopCh, informerSyncTimeout)
+}
+
+func (s *serviceCreationLatencyMeasurement) waitForReady() error {
+	for {
+		created, ipAssigned, reachable := 0, 0, 0
+		select {
+		case <-time.After(s.waitTimeout):
+			return fmt.Errorf("waiting for services timeout")
+		case <-time.After(defaultCheckInterval):
+			func() {
+				reachable = s.creationTimes.Count(reachabilityPhase)
+				ipAssigned = s.creationTimes.Count(ipAssigningPhase)
+				created = s.creationTimes.Count(creatingPhase)
+
+			}()
+			klog.Infof("%s: %d created, %d ipAssigned, %d reachable", s, created, ipAssigned, reachable)
+			if created == reachable {
+				return nil
+			}
+		}
+	}
+}
+
+func (s *serviceCreationLatencyMeasurement) gather() ([]measurement.Summary, error) {
+	klog.Infof("%s: gathering service created latency measurement...", s)
+	if !s.isRunning {
+		return nil, fmt.Errorf("metric %s has not been started", s)
+	}
+
+	serviceCreationLatency := s.creationTimes.CalculateTransitionsLatency(map[string]measurementutil.Transition{
+		"create_to_assigned": {
+			From: creatingPhase,
+			To:   ipAssigningPhase,
+		},
+		"assigned_to_available": {
+			From: ipAssigningPhase,
+			To:   reachabilityPhase,
+		},
+		"create_to_available": {
+			From: creatingPhase,
+			To:   reachabilityPhase,
+		},
+	})
+
+	content, err := util.PrettyPrintJSON(measurementutil.LatencyMapToPerfData(serviceCreationLatency))
+	if err != nil {
+		return nil, err
+	}
+	summary := measurement.CreateSummary(serviceCreationLatencyName, "json", content)
+	return []measurement.Summary{summary}, err
```

Comment:
s/err/nil/ 
[there is no way for it to be non-nil in this codepath]

Reply:
Done.